pkg/client/typed/dynamic/client_pool.go

```diff
@@ -58,13 +59,17 @@ func NewClientPool(config *restclient.Config, apiPathResolverFunc APIPathResolve
 	}
 }
 
-// ClientForGroupVersion returns a client for the specified groupVersion, creates one if none exists
-func (c *clientPoolImpl) ClientForGroupVersion(groupVersion unversioned.GroupVersion) (*Client, error) {
+// ClientForGroupVersion returns a client for the specified groupVersion,
+// creates one if none exists. The returned client will have the passed-in
+// parameterCodec. If the client exists already, its parameter codec will be
+// overwritten.
+func (c *clientPoolImpl) ClientForGroupVersion(groupVersion unversioned.GroupVersion, parameterCodec runtime.ParameterCodec) (*Client, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
 	// do we have a client already configured?
 	if existingClient, found := c.clients[groupVersion]; found {
+		existingClient.parameterCodec = parameterCodec
```

Comment:
Again, this allows something to modify a client that something else could be using.